### PR TITLE
Upgrade to phpspec 3.0

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -81,14 +81,13 @@
...
     "require-dev": {
-        "akeneo/phpspec-skip-example-extension": "~1.2",
+        "akeneo/phpspec-skip-example-extension": "^2.0",
         "behat/behat": "~3.0",
         "behat/mink": "~1.6",
         "behat/mink-browserkit-driver": "~1.2",
         "behat/mink-extension": "~2.0",
         "behat/mink-selenium2-driver": "~1.2",
         "behat/symfony2-extension": "~2.0",
-        "bossa/phpspec2-expect": "^1.0",
```
Comment (ciaranmcnulty) [inline composer.json]: I believe this is updated to support PhpSpec 3 soon, FYI

File: src/Sylius/Behat/Context/Domain/ProductContext.php [hunk 2]
```diff
@@ -153,14 +154,19 @@ public function iDeleteTheProduct(ProductInterface $product)
...
     }
 
     /**
      * @Then /^there should be no variants of (this product) in the product catalog$/
      */
     public function thereAreNoVariants(ProductInterface $product)
     {
-        expect($this->productVariantRepository->findBy(['object' => $product]))->toBe([]);
+        $variants = $this->productVariantRepository->findBy(['object' => $product]);
+
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Domain/ProductContext.php]: Redundant

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCountriesContext.php [hunk 3]
```diff
@@ -176,7 +176,13 @@ public function thisCountryShouldBeDisabled(CountryInterface $country)
...
     public function iShouldNotBeAbleToChoose($name)
     {
-        expect($this->createPage)->toThrow(ElementNotFoundException::class)->during('chooseName', [$name]);
+        try {
+            $this->createPage->chooseName($name);
+        } catch (ElementNotFoundException $exception) {
+            return;
+        }
+
+        throw new \DomainException('Choose name should throw an exception!');
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCountriesContext.php]: Expect is great for "throws", hope we get it back soon.

How about putting "throw new" in try block?

**Comment by pjedrzejewski** [comment]
Thanks Kamil! 👍

**Pull request merged**